### DISCUSSION TITLE
cps测试问题

### DISCUSSION THREAD

**Discussion by Pars1f4l** [baidu/dperf#450] [Q&A]
client配置
```
mode            client
cpu             0
tx_burst        128
launch_num      10
slow_start      30
payload_size    1
duration        2m
cps             600k
fast_close
port            0000:01:00.0    192.168.10.9    192.168.10.1
client          192.168.10.9    50
server          172.16.2.1      1
listen          443             50
```
server配置
```
mode        server
cpu         0
tx_burst    128
payload_size 1
duration    10m
port        0000:01:00.0      192.168.10.8   192.168.10.1 00:00:5e:00:01:01 
client      172.16.3.1        50
server      192.168.10.8      1
listen      8001              50
```

测试中发现运行一段时间后，不管cps设置为多少，dpvs都会出现错误如下
<img width="392" alt="image" src="https://github.com/baidu/dperf/assets/56541921/9ebe3157-802b-4684-afc4-114088bbe877">
同时cps变化情况
<img width="111" alt="image" src="https://github.com/baidu/dperf/assets/56541921/10d655b2-d6ec-4746-8cb8-51d118471d70">
可以看到cps在维持60w几秒后由于dpvs的报错问题，cps开始上下起伏，同时用40w的cps进行测试也能够发现cps稳定在40w一段时间后会有起伏，只是稳定时间更长，同时client和server端会有重传现象，说明性能下降
<img width="1031" alt="image" src="https://github.com/baidu/dperf/assets/56541921/adbe5f48-cddd-44e6-9802-a960c412c62b">

疑问：
1.如果测试dpvs的cps性能，选取cps指标的话是得cps稳定保持还是说稳定保持一段时间即可？
2.client和server端配置还能够优化吗？已经使用fast_close参数了
3.dpvs设置上有无能够优化的地方？

**Comment by pengjianzhang** [comment]
1. 建议稳定1小时
2. tx burst 4
3. 建议去dpvs社区询问，尝试增加大页，并把连接老化时间调小
